platform-hub-api/app/services/kubernetes/token_claim_service.rb

```diff
@@ -0,0 +1,75 @@
+module Kubernetes
+  module TokenClaimService
+    extend TokenLookup
+    extend self
+
+    module Errors
+      class TokenNotFound < StandardError; end
+      class TokenAlreadyDefinedForCluster < StandardError; end
+      class TokenInvalid < StandardError; end
+      class TokenAlreadyClaimed < StandardError; end
+    end
+
+    def claim_token(user, token)
```

Comment:
Should wrap everything in here in a transaction to avoid inconsistent data on failure.